code/modules/interaction_tests/interaction_test.dm

```diff
@@ -0,0 +1,34 @@
+/datum/interaction_test
```

Comment:
I think this is 100% something we need, but it should be a subtype of the unit tests I think, and use similar method's to TG's form of allocating. For example, here's a sample of `combat.dm` from TG. They have allocation on all their unit tests.
```
/datum/unit_test/harm_melee/Run()
	var/mob/living/carbon/human/tider = allocate(/mob/living/carbon/human/consistent)
	var/mob/living/carbon/human/victim = allocate(/mob/living/carbon/human/consistent)
	var/obj/item/storage/toolbox/toolbox = allocate(/obj/item/storage/toolbox)

	tider.put_in_active_hand(toolbox, forced = TRUE)
	tider.set_combat_mode(TRUE)
	victim.attackby(toolbox, tider)

	TEST_ASSERT(victim.getBruteLoss() > 0, "Victim took no brute damage after being hit by a toolbox")
```

Reply:
I've integrated the allocator into the puppeteer with the new test setup, PTAL.